R/dates_and_times.R

```diff
@@ -0,0 +1,76 @@
+#' Fix UTC offset strings
+#' 
+#' UTC offsets can be formatted in multiple ways (e.g. -07, -07:00, -0700) and R often struggles to parse these offsets. This function takes date/time strings with valid UTC offsets, and formats them so that they are consistent and readable by R.
+#'
+#' @param datetime_strings Character vector of dates in ISO 8601 format
+#'
+#' @return datetime_strings with UTC offsets consistently formatted to four digits (e.g. "2023-11-16T03:32:49-0700").
+#' @export
+#'
+#' @examples
+#' datetimes <- c("2023-11-16T03:32:49+07:00","2023-11-16T03:32:49-07","2023-11-16T03:32:49","2023-11-16T03:32:49Z")
+#' fix_utc_offset(datetimes)  # returns c("2023-11-16T03:32:49+0700", "2023-11-16T03:32:49-0700", "2023-11-16T03:32:49", "2023-11-16T03:32:49+0000") and warns about missing offset (see third element)
+#' 
+fix_utc_offset <- function(datetime_strings) {
+  datetime_strings <- stringr::str_replace_all(datetime_strings, "[−‐‑‒–—―﹘﹣－]", "-") # replace every possible type of dash with a regular minus sign
```

Comment:
looks like non-ascii characters, likely in this line mucking up the package build.